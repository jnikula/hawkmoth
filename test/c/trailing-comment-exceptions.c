typedef struct {
    int a;
    int b; 
} multiline; /**< Multiline construct not documented */

int a;
/**< trailing comment must start on the same line as the construct */

/** This leading comment should still apply */
int b; /**< trailing comment does not merge with leading comment */

int multiline_comment; /**<
                        * trailing comment that has multiple lines
                        * should not be applied
                        */

void short_function(void) /**< no comments in middle of line */ {return;}

void short_function_2( /**< no comments in middle of construct */ void) {return;}
