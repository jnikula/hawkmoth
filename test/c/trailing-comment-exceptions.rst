
.. c:var:: int b

   This leading comment should still apply

