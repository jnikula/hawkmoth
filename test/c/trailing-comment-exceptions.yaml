directives:
- domain: c
  directive: autodoc
  arguments:
  - trailing-comment-exceptions.c
expected: trailing-comment-exceptions.rst
errors: trailing-comment-exceptions.stderr
