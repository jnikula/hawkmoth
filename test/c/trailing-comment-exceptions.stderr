WARNING: trailing-comment-exceptions.c:4: trailing comment for multiline construct was dropped.
WARNING: trailing-comment-exceptions.c:7: trailing comment on separate line from documented construct was dropped.
WARNING: trailing-comment-exceptions.c:10: multiple comments for a single construct; only first was kept.
WARNING: trailing-comment-exceptions.c:12: multiline trailing comment was dropped.
WARNING: trailing-comment-exceptions.c:17: trailing comment was not the last token on a line.
WARNING: trailing-comment-exceptions.c:19: trailing comment in the middle of a construct was dropped.
